--- a/CMakeLists.txt	2016-05-30 22:29:20.953399971 +0200
+++ b/CMakeLists.txt	2016-05-30 22:29:37.766453575 +0200
@@ -241,7 +241,6 @@
 install( TARGETS tint2 DESTINATION bin )
 install( FILES tint2.svg DESTINATION ${DATADIR}/icons/hicolor/scalable/apps )
 install( FILES tint2.desktop DESTINATION ${DATADIR}/applications )
-install( CODE "execute_process(COMMAND gtk-update-icon-cache -f -t ${DATADIR}/icons/hicolor WORKING_DIRECTORY ${CMAKE_INSTALL_PREFIX})" )
 install( FILES themes/tint2rc DESTINATION ${SYSCONFDIR}/xdg/tint2 )
 install( FILES default_icon.png DESTINATION ${DATADIR}/tint2 )
 install( FILES AUTHORS ChangeLog README.md doc/tint2.md doc/manual.html doc/readme.html DESTINATION ${DOCDIR} )
--- a/src/tint2conf/CMakeLists.txt	2016-05-30 22:33:13.602860462 +0200
+++ b/src/tint2conf/CMakeLists.txt	2016-05-30 22:33:34.115852109 +0200
@@ -72,4 +72,3 @@
 install( TARGETS tint2conf DESTINATION bin )
 install( FILES tint2conf.svg DESTINATION ${DATADIR}/icons/hicolor/scalable/apps )
 install( FILES tint2conf.desktop DESTINATION ${DATADIR}/applications )
-install( CODE "execute_process(COMMAND gtk-update-icon-cache -f -t ${DATADIR}/icons/hicolor WORKING_DIRECTORY ${CMAKE_INSTALL_PREFIX})" )
