--- /usr/share/applications/viber.desktop.old	2014-04-03 12:24:38.000000000 +0400
+++ /usr/share/applications/viber.desktop	2014-04-03 12:25:03.000000000 +0400
@@ -1,10 +1,10 @@
 [Desktop Entry]
 Name=Viber
 Comment=Viber VoIP and messenger
-Exec=/usr/share/viber/Viber.sh
+Exec=/opt/viber/Viber.sh
 Icon=/usr/share/pixmaps/viber.png
 Terminal=false
 Type=Application
 Encoding=UTF-8
-Categories=Network;Application;
-Path=
+Categories=Network;
+Path=/opt/viber
