--- setup.py.old	2012-02-05 08:59:19.000000000 +0400
+++ setup.py	2013-08-01 18:41:11.388812135 +0400
@@ -8,23 +8,12 @@
     use_setuptools()
     from setuptools import setup, find_packages
 
-from distutils.command.build import build as _build
-from babel.messages import frontend as babel
-
 from turpial import VERSION
 
 LONG_DESCRIPTION = """
-Turpial is a light, fast and beautiful microblogging client for GNU/Linux, 
-written in Python and fully functional
+Turpial is a light, fast and beautiful microblogging client written in Python
 """
 
-class build(_build):
-    sub_commands = [('compile_catalog', None), ] + _build.sub_commands
-
-    def run(self):
-        """Run all sub-commands"""
-        _build.run(self)
-
 # TODO: Maybe find some better ways to do this
 # looking distutils's copy_tree method
 data_files=[
@@ -34,53 +23,39 @@
     ('share/applications', ['turpial.desktop']),
     ('share/doc/turpial', ['ChangeLog', 'README.rst']),
 ]
-'''
-pattern = re.compile('turpial/i18n/')
-for root, dirs, files in os.walk(os.path.join('turpial', 'i18n')):
-    for filename in files:
-        if filename.endswith('.mo'):
-            fullpath = os.path.join(root, filename)
-            dest = os.path.join('/', 'usr', 'share', 'locale', re.sub(pattern, '', root))
-            data_files.append((dest, [fullpath]))
-'''
+
 setup(name="turpial",
-      version=VERSION,
-      description="A light, beautiful and functional microblogging client",
-      long_description=LONG_DESCRIPTION,
-      author="Wil Alvarez",
-      author_email="wil.alejandro@gmail.com",
-      maintainer="Wil Alvarez",
-      maintainer_email="wil.alejandro@gmail.com",
-      url="http://turpial.org.ve",
-      download_url="http://turpial.org.ve/downloads",
-      license="GPLv3",
-      keywords='twitter identi.ca microblogging turpial',
-      classifiers=[
-        "Development Status :: 5 - Production/Stable",
-        "Environment :: X11 Applications :: GTK",
-        "Intended Audience :: End Users/Desktop",
-        "License :: OSI Approved :: GNU General Public License (GPL)",
-        "Operating System :: POSIX :: Linux",
-        "Programming Language :: Python",
-        "Topic :: Communications"
+    version=VERSION,
+    description="A light, beautiful and functional microblogging client",
+    long_description=LONG_DESCRIPTION,
+    author="Wil Alvarez",
+    author_email="wil.alejandro@gmail.com",
+    maintainer="Wil Alvarez",
+    maintainer_email="wil.alejandro@gmail.com",
+    url="http://turpial.org.ve",
+    download_url="http://turpial.org.ve/downloads",
+    license="GPLv3",
+    keywords='twitter identi.ca microblogging turpial',
+    classifiers=[
+      "Development Status :: 5 - Production/Stable",
+      "Environment :: X11 Applications :: GTK",
+      "Intended Audience :: End Users/Desktop",
+      "License :: OSI Approved :: GNU General Public License (GPL)",
+      "Operating System :: POSIX :: Linux",
+      "Programming Language :: Python",
+      "Topic :: Communications"
+    ],
+    include_package_data=True,
+    packages=find_packages(),
+    package_data={
+      'turpial': ['data/pixmaps/*', 'data/sounds/*', 'data/layout/*.template',
+      'data/layout/*.partial', 'data/layout/js/*', 'data/layout/css/*']
+    },
+    entry_points={
+      'console_scripts': [
+          'turpial = turpial.main:main',
+          'turpial-unity-daemon = turpial.ui.unity.daemon:main',
       ],
-      packages=find_packages(),
-      package_data={
-        'turpial': ['data/pixmaps/*', 'data/sounds/*', 'data/themes/default/*', 
-            'data/layout/*.template', 'data/layout/*.partial', 
-            'data/layout/js/*', 'data/layout/css/*']
-      },
-      entry_points={
-        'console_scripts': [
-            'turpial = turpial.main:Turpial',
-        ],
-      },
-      cmdclass={
-        'build': build,
-        'compile_catalog': babel.compile_catalog,
-        'extract_messages': babel.extract_messages,
-        'init_catalog': babel.init_catalog,
-        'update_catalog': babel.update_catalog,
-      },
-      data_files=data_files,
+    },
+    data_files=data_files,
 )
