diff -Nuar suricata-3.2.1-old/src/Makefile.am suricata-3.2.1/src/Makefile.am
--- suricata-3.2.1-old/src/Makefile.am	2017-02-15 10:54:12.000000000 +0300
+++ suricata-3.2.1/src/Makefile.am	2017-03-14 09:02:14.306193704 +0300
@@ -471,8 +471,6 @@
 PTXS += $(suricata_CUDA_KERNELS:.cu=.ptx_sm_35)
 PTXS += $(suricata_CUDA_KERNELS:.cu=.ptx_sm_37)
 PTXS += $(suricata_CUDA_KERNELS:.cu=.ptx_sm_50)
-PTXS += $(suricata_CUDA_KERNELS:.cu=.ptx_sm_52)
-PTXS += $(suricata_CUDA_KERNELS:.cu=.ptx_sm_53)
 
 .cu.ptx_sm_20:
 	$(NVCC) $(NVCCFLAGS) -o $@ -arch=sm_20 -ptx $<
@@ -495,12 +493,6 @@
 .cu.ptx_sm_50:
 	$(NVCC) $(NVCCFLAGS) -o $@ -arch=sm_50 -ptx $<
 
-.cu.ptx_sm_52:
-	$(NVCC) $(NVCCFLAGS) -o $@ -arch=sm_52 -ptx $<
-
-.cu.ptx_sm_53:
-	$(NVCC) $(NVCCFLAGS) -o $@ -arch=sm_53 -ptx $<
-
 cuda-ptxdump.h: $(PTXS)
 	$(PYTHON) ptxdump.py cuda-ptxdump $(PTXS)
 
diff -Nuar suricata-3.2.1-old/src/Makefile.in suricata-3.2.1/src/Makefile.in
--- suricata-3.2.1-old/src/Makefile.in	2017-02-15 10:54:31.000000000 +0300
+++ suricata-3.2.1/src/Makefile.in	2017-03-14 09:01:50.562095593 +0300
@@ -1055,9 +1055,7 @@
 @BUILD_CUDA_TRUE@	$(suricata_CUDA_KERNELS:.cu=.ptx_sm_32) \
 @BUILD_CUDA_TRUE@	$(suricata_CUDA_KERNELS:.cu=.ptx_sm_35) \
 @BUILD_CUDA_TRUE@	$(suricata_CUDA_KERNELS:.cu=.ptx_sm_37) \
-@BUILD_CUDA_TRUE@	$(suricata_CUDA_KERNELS:.cu=.ptx_sm_50) \
-@BUILD_CUDA_TRUE@	$(suricata_CUDA_KERNELS:.cu=.ptx_sm_52) \
-@BUILD_CUDA_TRUE@	$(suricata_CUDA_KERNELS:.cu=.ptx_sm_53)
+@BUILD_CUDA_TRUE@	$(suricata_CUDA_KERNELS:.cu=.ptx_sm_50)
 @BUILD_CUDA_TRUE@CLEANFILES = $(PTXS) cuda-ptxdump.h
 
 # default CFLAGS
@@ -1070,7 +1068,7 @@
 	$(MAKE) $(AM_MAKEFLAGS) all-am
 
 .SUFFIXES:
-.SUFFIXES: .c .cu .lo .o .obj .ptx_sm_20 .ptx_sm_21 .ptx_sm_30 .ptx_sm_32 .ptx_sm_35 .ptx_sm_37 .ptx_sm_50 .ptx_sm_52 .ptx_sm_53
+.SUFFIXES: .c .cu .lo .o .obj .ptx_sm_20 .ptx_sm_21 .ptx_sm_30 .ptx_sm_32 .ptx_sm_35 .ptx_sm_37 .ptx_sm_50
 $(srcdir)/Makefile.in:  $(srcdir)/Makefile.am  $(am__configure_deps)
 	@for dep in $?; do \
 	  case '$(am__configure_deps)' in \
@@ -1854,12 +1852,6 @@
 @BUILD_CUDA_TRUE@.cu.ptx_sm_50:
 @BUILD_CUDA_TRUE@	$(NVCC) $(NVCCFLAGS) -o $@ -arch=sm_50 -ptx $<
 
-@BUILD_CUDA_TRUE@.cu.ptx_sm_52:
-@BUILD_CUDA_TRUE@	$(NVCC) $(NVCCFLAGS) -o $@ -arch=sm_52 -ptx $<
-
-@BUILD_CUDA_TRUE@.cu.ptx_sm_53:
-@BUILD_CUDA_TRUE@	$(NVCC) $(NVCCFLAGS) -o $@ -arch=sm_53 -ptx $<
-
 @BUILD_CUDA_TRUE@cuda-ptxdump.h: $(PTXS)
 @BUILD_CUDA_TRUE@	$(PYTHON) ptxdump.py cuda-ptxdump $(PTXS)
 
