--- include/host_config.h.old	2017-03-14 07:00:23.642220759 +0300
+++ include/host_config.h	2017-03-14 07:00:40.432712299 +0300
@@ -77,11 +77,11 @@
 
 #if defined(__GNUC__)
 
-#if __GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ > 8)
+#if __GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ > 9)
 
 #error -- unsupported GNU version! gcc 4.9 and up are not supported!
 
-#endif /* __GNUC__> 4 || (__GNUC__ == 4 && __GNUC_MINOR__ > 8) */
+#endif /* __GNUC__> 4 || (__GNUC__ == 4 && __GNUC_MINOR__ > 9) */
 
 #endif /* __GNUC__ */
 
