diff -Naur osmo-0.4.2-old/configure.ac osmo-0.4.2-new/configure.ac
--- osmo-0.4.2-old/configure.ac	2017-11-05 19:13:09.000000000 +0500
+++ osmo-0.4.2-new/configure.ac	2017-12-05 04:13:18.399869058 +0500
@@ -151,13 +151,11 @@
     libwebkit="no"
 fi
 
-contacts="yes"; 
 htmlrender="no";
 if test "x$have_libwebkit" = "xfalse"; then 
         contacts="no"; 
 else
     htmlrender="webkit";
-    contacts="yes"; 
     AC_DEFINE([HAVE_LIBWEBKIT], [1], [Definded to 1 if compile with libwebkit support])
     AC_DEFINE([HTMLRENDER], ["webkit"], [HTML render type])
 fi
