diff -Nuar osmo-0.4.2-orig/configure.ac osmo-0.4.2/configure.ac
--- osmo-0.4.2-orig/configure.ac	2017-11-05 17:13:09.000000000 +0300
+++ osmo-0.4.2/configure.ac	2018-12-26 11:41:43.447242474 +0300
@@ -143,7 +143,7 @@
 fi
 
 # Checks for libwebkit
-PKG_CHECK_MODULES(LIBWEBKIT, webkit2gtk-4.0 >= 2.4.0, have_libwebkit=true,
+PKG_CHECK_MODULES(LIBWEBKIT, webkit2gtk-4.0 >= 2.8.0, have_libwebkit=true,
     have_libwebkit=false)
 if test "x$have_libwebkit" = "xtrue"; then
     libwebkit="yes"
diff -Nuar osmo-0.4.2-orig/src/calendar.c osmo-0.4.2/src/calendar.c
--- osmo-0.4.2-orig/src/calendar.c	2017-09-22 14:07:54.000000000 +0300
+++ osmo-0.4.2/src/calendar.c	2018-12-26 11:44:26.827256633 +0300
@@ -1071,13 +1071,13 @@
 
 #ifdef CONTACTS_ENABLED
         /* check contacts */
-        i = check_add_contacts (date, TRUE, appGUI);
+        i = check_add_contacts (date, TRUE, NULL, appGUI);
 
         if (i) {
             g_snprintf (tmpbuf, BUFFER_SIZE, "%s:\n", _("Birthday"));
             gtk_text_buffer_insert_with_tags_by_name (appGUI->cal->day_desc_text_buffer,
                                                       &appGUI->cal->day_desc_iter, tmpbuf, -1, "bold", NULL);
-            check_add_contacts (date, FALSE, appGUI);
+            check_add_contacts (date, FALSE, NULL, appGUI);
             gtk_text_buffer_insert (appGUI->cal->day_desc_text_buffer, &appGUI->cal->day_desc_iter, "\n", -1);
         }
 #endif  /* CONTACTS_ENABLED */
diff -Nuar osmo-0.4.2-orig/src/contacts_items.c osmo-0.4.2/src/contacts_items.c
--- osmo-0.4.2-orig/src/contacts_items.c	2017-09-22 14:07:54.000000000 +0300
+++ osmo-0.4.2/src/contacts_items.c	2018-12-26 11:46:49.581269004 +0300
@@ -678,7 +678,7 @@
         if (i == COLUMN_FIRST_NAME) {
             appGUI->cnt->next_field_button[i] = gtk_button_new_from_icon_name ("go-down", GTK_ICON_SIZE_BUTTON);
             gtk_widget_show (appGUI->cnt->next_field_button[i]);
-            gtk_widget_set_can_focus(appGUI->cnt->next_field_button[i], false);
+            gtk_widget_set_can_focus(appGUI->cnt->next_field_button[i], FALSE);
         }
 
         gtk_widget_show (appGUI->cnt->contact_entries[i]);
diff -Nuar osmo-0.4.2-orig/src/main.c osmo-0.4.2/src/main.c
--- osmo-0.4.2-orig/src/main.c	2018-12-26 11:37:04.655218314 +0300
+++ osmo-0.4.2/src/main.c	2018-12-26 11:40:34.995236542 +0300
@@ -349,8 +349,12 @@
     g_free (cal);
     g_free (appGUI);
     g_free(cmd_cfg_path);
+#ifdef CONTACTS_ENABLED
     g_free(cmd_mutt_query);
+#endif  /* CONTACTS_ENABLED */
+#ifdef HAVE_LIBWEBKIT
     g_free(cmd_stylesheet_file);
+#endif  /* HAVE_LIBWEBKIT */
 #ifndef WIN32
     g_free (s_lock);
 #endif /* WIN32 */
