diff -burN android-file-transfer-linux-2.2.orig/CMakeLists.txt android-file-transfer-linux-2.2/CMakeLists.txt
--- android-file-transfer-linux-2.2.orig/CMakeLists.txt	2015-05-18 05:09:33.000000000 +0800
+++ android-file-transfer-linux-2.2/CMakeLists.txt	2015-05-20 20:37:39.446876364 +0800
@@ -6,11 +6,15 @@
 include(CheckFunctionExists)
 
 find_package ( Threads )
-pkg_check_modules ( FUSE fuse )
 
-if (FUSE_FOUND)
+option(ENABLE_FUSE "Enable fuse interface support" OFF)
+if (ENABLE_FUSE)
+	pkg_check_modules ( FUSE fuse )
+
+	if (FUSE_FOUND)
 	message(STATUS "fuse found, building mount helper")
 	add_definitions(${FUSE_CFLAGS} -DFUSE_USE_VERSION=26)
+	endif()
 endif()
 
 option(BUILD_QT_UI "Build reference Qt application" ON)
@@ -87,12 +91,15 @@
 
 add_executable(aft-mtp-cli mtp/cli.cpp)
 target_link_libraries(aft-mtp-cli ${MTP_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT})
+install(PROGRAMS ${CMAKE_CURRENT_BINARY_DIR}/aft-mtp-cli DESTINATION bin)
 
-if (FUSE_FOUND)
+if (ENABLE_FUSE)
+	if (FUSE_FOUND)
 	add_executable(aft-mtp-mount mtp/fuse.cpp)
 	target_link_libraries(aft-mtp-mount ${MTP_LIBRARIES} ${FUSE_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT})
 	install(PROGRAMS ${CMAKE_CURRENT_BINARY_DIR}/aft-mtp-mount DESTINATION bin)
-endif()
+	endif (FUSE_FOUND)
+endif (ENABLE_FUSE)
 
 if (BUILD_QT_UI)
 	add_subdirectory(qt)
diff -burN android-file-transfer-linux-2.2.orig/qt/CMakeLists.txt android-file-transfer-linux-2.2/qt/CMakeLists.txt
--- android-file-transfer-linux-2.2.orig/qt/CMakeLists.txt	2015-05-18 05:09:33.000000000 +0800
+++ android-file-transfer-linux-2.2/qt/CMakeLists.txt	2015-05-20 20:40:14.172887578 +0800
@@ -1,7 +1,17 @@
 project(mtp-qt)
 
-find_package(Qt5Widgets QUIET)
-find_package(Qt4 QUIET)
+option(USE_QT4 "Build Qt4 GUI" OFF)
+option(USE_QT5 "Build Qt5 GUI" ON)
+
+if (NOT USE_QT4 AND NOT USE_QT5)
+	message(FATAL_ERROR "Neither Qt4 nor Qt5 enabled")
+endif()
+
+if (USE_QT5)
+	find_package(Qt5Widgets REQUIRED)
+elseif (USE_QT4)
+	find_package(Qt4 REQUIRED)
+endif()
 
 set(SOURCES main.cpp
 	mainwindow.cpp
@@ -31,22 +41,30 @@
 	renamedialog.ui
 )
 
-if (Qt5Widgets_FOUND)
-	message(STATUS "Using Qt5")
-	qt5_wrap_cpp(HEADERS_MOC ${HEADERS})
-	qt5_wrap_ui(FORMS_HEADERS ${FORMS})
-	qt5_add_resources(RESOURCES android-file-transfer.qrc)
-	include_directories(${Qt5Widgets_INCLUDE_DIRS})
-	set(EXTRA_QT_LINK ${Qt5Widgets_LIBRARIES})
-elseif(Qt4_FOUND OR QT_FOUND OR QT4_FOUND)
+if (USE_QT4)
+	if(Qt4_FOUND OR QT_FOUND OR QT4_FOUND)
 	message(STATUS "Using Qt4")
 	qt4_wrap_cpp(HEADERS_MOC ${HEADERS})
 	qt4_wrap_ui(FORMS_HEADERS ${FORMS})
 	qt4_add_resources(RESOURCES android-file-transfer.qrc)
 	include(${QT_USE_FILE})
 	set(EXTRA_QT_LINK ${QT_LIBRARIES})
-else()
-	message(FATAL_ERROR "No Qt found")
+	else()
+		message(FATAL_ERROR "Qt4 enabled but not found")
+	endif()
+endif()
+
+if (USE_QT5)
+	if (Qt5Widgets_FOUND)
+		message(STATUS "Using Qt5")
+		qt5_wrap_cpp(HEADERS_MOC ${HEADERS})
+		qt5_wrap_ui(FORMS_HEADERS ${FORMS})
+		qt5_add_resources(RESOURCES android-file-transfer.qrc)
+		include_directories(${Qt5Widgets_INCLUDE_DIRS})
+		set(EXTRA_QT_LINK ${Qt5Widgets_LIBRARIES})
+	else()
+		message(FATAL_ERROR "Qt5 enabled but not found")
+	endif()
 endif()
 
 add_definitions(${QT_DEFINITIONS})
