--- usr/share/applications/pburn.desktop.original	2014-10-28 14:12:59.703006612 +0300
+++ usr/share/applications/pburn.desktop	2014-10-28 14:12:22.979010052 +0300
@@ -7,10 +7,11 @@
 Name[hu_HU]=Pburn CD/DVD/Blu-ray író
 Name[ja]=Pburn CD/DVD/Blu-ray書き込み
 Name[nb_NO]=Pburn Brenn CD/DVD/Blu-ray
-Icon=/usr/share/icons/hicolor/scalable/apps/pburn.svg
+Name[ru]=Запись CD/DVD/Blu-ray
+Icon=pburn
 Comment=Pburn CD/DVD/Blu-ray writer
 Exec=pburn
 Terminal=false
 Type=Application
-Categories=DiskBurning
+Categories=AudioVideo;DiscBurning;
 GenericName=Pburn CD/DVD/Blu-ray writer
